### ISSUE TITLE
Newest Amplify framework does not build with Xcode 15.3 beta 2

### ISSUE THREAD

**Issue by ethan021021** [aws-amplify/amplify-swift#3499]
### Describe the bug

Exactly as the title says. Attached is a screenshot

<img width="391" alt="Screenshot 2024-02-06 at 12 53 10 PM" src="https://github.com/aws-amplify/amplify-swift/assets/15329658/6ed37f2a-fc21-41a2-93c3-533f505e9d08">

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Open Xcode 15.3 beta 2
2. Try to compile iOS application with amplify-swift framework 2.25.6
```

### Expected behavior

The project to build

### Amplify Framework Version

2.25.6

### Amplify Categories

DataStore

### Dependency manager

Swift PM

### Swift version

5.6

### CLI version

5.10

### Xcode version

15.3 beta 2

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

17.2

### Device

iPhone 15 Pro

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by lawmicha** [comment]
Thanks for opening the issue. Our team will investigate and provide an update.

**Comment by thisisabhash** [comment]
Hello @ethan021021, 
We have a build fix in the branch : https://github.com/aws-amplify/amplify-swift/tree/fix/build-xcode-15-3-beta-2
Let us know if it works for you.

**Comment by ethan021021** [comment]
Awesome, thanks for the quick fix!

**Comment by ethan021021** [comment]
Hey all,

Can confirm the fix works. Thank you!

**Comment by lawmicha** [comment]
Thanks for letting us know, we'll be releasing this shortly in 2.25.7

**Comment by ethan021021** [comment]
@lawmicha 

Apologies as I spoke too soon.

It looks like in the PR no changes to Attempt.swift were made so the build is still failing

**Comment by thisisabhash** [comment]
Hello @ethan021021 - Can you try out this branch in your project - 
https://github.com/aws-amplify/amplify-swift/tree/fix/build-xcode15_3_2

**Comment by ethan021021** [comment]
@thisisabhash can confirm this compiles now. Thanks for the fix!

**Comment by lawmicha** [comment]
Thanks for confirming, https://github.com/aws-amplify/amplify-swift/pull/3506 should be released in either 2.26.1 or 2.26.2, will keep you posted

**Comment by lawmicha** [comment]
This has been released in 2.26.1 https://github.com/aws-amplify/amplify-swift/releases/tag/2.26.1
